pkg/admission/validator/shoot.go

```diff
@@ -179,6 +179,9 @@ func (s *shoot) validateUpdate(oldShoot, shoot *core.Shoot, cloudProfile *garden
 
 	allErrs = append(allErrs, s.validateShoot(shoot, oldInfraConfig, infraConfig, cloudProfile, cpConfig)...)
 
+	// TODO(dkistner) remove this once csi-driver async operation issues are resolved.
+	allErrs = append(allErrs, azurevalidation.ValidateUpgradeV120ToV121(shoot, oldShoot, specPath.Child("kubernetes", "version"))...)
```

Comment:
We could also add a note to `docs/usage-as-end-user.md#csi-volume-provisioners` for the end users describing what is the issue/risk and that it is still possible to upgrade using the annotation `shoot.gardener.cloud/force-version-upgrade`.